crates/cxx-qt-gen/src/generator/rust/signals.rs

```diff
@@ -65,14 +65,18 @@ pub fn generate_rust_signal(
         .cloned()
         .map(|parameter| match parameter {
             FnArg::Typed(pat_type) => syn_type_cxx_bridge_to_qualified(&pat_type.ty, type_names),
-            _ => unreachable!("should only have typed no receiver"),
+            _ => {
+                // CODECOV_EXCLUDE_START
+                unreachable!("should only have typed no receiver")
+                // CODECOV_EXCLUDE_STOP
+            }
         })
         .collect::<Result<_>>()?;
 
     let self_type_cxx = if signal.mutable {
         parse_quote! { Pin<&mut #qobject_name_rust> }
     } else {
-        parse_quote! { &#qobject_name_rust }
+        parse_quote! { &#qobject_name_rust } // Signals cannot be mutable
```

Comment:
Think that comment should be `Signals cannot be immutable` ? As they have to be mutable :-) (and this is another unreachable!() place i assume?)

Reply:
yeah it is for now, I was wondering if that indicates a bigger refactor though, that we have a signal mutability field and options handling it, but they can never be immutable

Reply:
For now we are leaving the door open to having them be immutable too so just erroring in the parser for now, but i'm unsure if that works on the Qt side as you typically make them mutable in C++. This is something we need to check and decide before 1.0.